config/config_helpers.go

```diff
@@ -632,17 +633,26 @@ func getAWSAccountAlias(ctx *ParsingContext, l log.Logger) (string, error) {
 
 // Return the AWS account id associated to the current set of credentials
 func getAWSAccountID(ctx *ParsingContext, l log.Logger) (string, error) {
+	l.Tracef("Invoking get_aws_account_id()...")
+
 	awsConfig, err := awshelper.CreateAwsConfig(ctx.Context, l, nil, ctx.TerragruntOptions)
 	if err != nil {
+		l.Errorf("get_aws_account_id(): failed to create AWS config: %v", err)
```

Comment:
Please remove this. The error should be propagated correctly to make it clear what happened.

Consider wrapping the error below instead.

```go
fmt.Errorf("failed to create AWS config for get_aws_account_id(): %w", err)
```